pkg/gitprovider/bitbucketserver.go

```diff
@@ -353,6 +353,68 @@ func (g *BitbucketServerGitProvider) getPrContext(staticContext *StaticGitContex
 	return &repo, nil
 }
 
+func (g *BitbucketServerGitProvider) GetBranchByCommit(staticContext *StaticGitContext) (string, error) {
+	if staticContext.Sha == nil || *staticContext.Sha == "" {
+		return *staticContext.Branch, nil
+	}
+
+	client, err := g.getApiClient()
+	if err != nil {
+		return "", err
+	}
+
+	branches, err := client.DefaultApi.GetBranches(staticContext.Owner, staticContext.Name, map[string]interface{}{})
+	if err != nil {
+		return "", err
+	}
+
+	branchList, err := bitbucketv1.GetBranchesResponse(branches)
+	if err != nil {
+		return "", err
+	}
+
+	var branchName string
+	for _, branch := range branchList {
+		if branch.LatestCommit == *staticContext.Sha {
+			branchName = branch.DisplayID
+			break
+		}
+
+		commits, err := client.DefaultApi.GetCommits(staticContext.Owner, staticContext.Name, map[string]interface{}{
```

Comment:
I can see this pattern repeated in some other providers as well.

You have a for loop going through branches, then you list commits but don't even have the branch in the query. In these situation, I don't think you even need to get the branch list. You can only list commits and then find the appropriate branch.
Also, here you have the `until` filter. Is there a `from` filter as well so this API returns only one entry?